### PR TITLE
fix: add support for RN 0.76

### PR DESCRIPTION
**Pull request by hannojg**
- Upgrades the example app to RN 0.76
- Adds a fix for android's CMakeLists to properly compile on RN 0.76
- Fixes the eslint setup to work with prettier

All tests on iOS and android are successful!

### CODE REVIEW

**Comment by hannojg** [comment]
CI still failing, hold on

File: src/plugin/__tests__/__snapshots__/index.spec.js.snap [hunk 1]
```diff
@@ -467,14 +465,13 @@ var Foo = function () {
       return _f;
     }()
   }]);
-  return Foo;
```
Comment (mrousavy) [inline src/plugin/__tests__/__snapshots__/index.spec.js.snap:470]: wait what?

**Review by mrousavy — APPROVED** [review]
lgtm!

File: src/plugin/__tests__/__snapshots__/index.spec.js.snap (hunk 1 above)
Comment (hannojg) [reply src/plugin/__tests__/__snapshots__/index.spec.js.snap:470]: it was because the babel plugin version changed i assume

**Comment by ska-anonymous** [comment]
The cmake fix is again not working with RN 0.77 and 0.78

**Comment by affansk** [comment]
77 and 78 cmake issue. not able to build

**Comment by scott-harrison** [comment]
any update for support for RN 0.77 or 0.78?

**Pull request merged**